widget/embedded/src/pages/ConfirmSwapPage.tsx

```diff
@@ -1,265 +1,372 @@
-import { navigationRoutes } from '../constants/navigationRoutes';
+import { SlippageWarningType } from '../types';
 import {
-  getKeplrCompatibleConnectedWallets,
-  getRequiredChains,
-  getSelectableWallets,
-  isExperimentalChain,
-} from '../utils/wallets';
+  numberToString,
+  secondsToString,
+  totalArrivalTime,
+} from '../utils/numbers';
 import {
   confirmSwapDisabled,
+  getConfirmSwapErrorMessage,
   getPercentageChange,
+  getRouteWarningMessage,
   getTotalFeeInUsd,
-  isValidCustomDestination,
 } from '../utils/swap';
-import { numberToString } from '../utils/numbers';
-import { useMetaStore } from '../store/meta';
-import { WalletType } from '@rango-dev/wallets-shared';
-import { useNavigateBack } from '../hooks/useNavigateBack';
-import {
-  Divider,
-  ConfirmSwap,
-  LoadingFailedAlert,
-  HighSlippageWarning,
-  PercentageChange,
-  RoutesOverview,
-  TokenPreview,
-} from '@rango-dev/ui';
-import { useManager } from '@rango-dev/queue-manager-react';
-import { useConfirmSwap } from '../hooks/useConfirmSwap';
-import { useSettingsStore } from '../store/settings';
-import { useUiStore } from '../store/ui';
-import { ConfirmSwapErrorTypes } from '../types';
-import { ConfirmSwapErrors } from '../components/ConfirmSwapErrors';
-import { ConfirmSwapWarnings } from '../components/ConfirmSwapWarnings';
-import { HIGH_SLIPPAGE } from '../constants/swapSettings';
-import { getBestRouteStatus } from '../utils/routing';
 
-export function ConfirmSwapPage({
-  customDestinationEnabled,
-}: {
+const Container = styled('div', {
+  position: 'relative',
+  width: '100%',
+  height: '593px',
+  '& .title': {
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    paddingBottom: '$10',
+  },
+  '& .icon': {
+    width: '$24',
+    height: '$24',
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  '& .buttons': {
+    width: '100%',
+    position: 'absolute',
+    bottom: '$0',
+    display: 'flex',
+    justifyContent: 'space-between',
+    filter: 'drop-shadow(white 0 -10px 10px)',
+  },
+  '& .confirm-button': {
+    flexGrow: 1,
+    paddingRight: '$10',
+  },
+});
+
+type PropTypes = {
   customDestinationEnabled?: boolean;
-}) {
+  supportedWallets: WidgetConfig['wallets'];
+  multiWallets: boolean;
+  config?: WidgetConfig;
+};
+
+export function ConfirmSwapPage(props: PropTypes) {
+  const { customDestinationEnabled, multiWallets, supportedWallets, config } =
+    props;
   const navigate = useNavigate();
-  const { navigateBackFrom } = useNavigateBack();
+  const [showWallets, setShowWallets] = useState(false);
+  const [dbErrorMessage, setDbErrorMessage] = useState<string>('');
+  const [showSlippageWarning, setShowSlippageWarning] = useState(false);
   const bestRoute = useBestRouteStore.use.bestRoute();
+  const [walletsConfirmed, setWalletsConfirmed] = useState(false);
+  const inputAmount = useBestRouteStore.use.inputAmount();
```

Comment:
Can you explain why we are call `useBestRouteStore.use.whatever()` separately? 
Could we use this approach:
```
const store = useBestRouteStore();
```
and whenever we need a value, we will call it in-place:
```
store.inputAmount()
```

I mean we can call the store and keep the store/state in a variable then in the rest of our program accessing the value directly and remove these:
```
 const outputAmount = useBestRouteStore.use.outputAmount();
  const inputUsdValue = useBestRouteStore.use.inputUsdValue();
  const outputUsdValue = useBestRouteStore.use.outputUsdValue();
  const setInputAmount = useBestRouteStore.use.setInputAmount();
  const selectedWallets = useBestRouteStore.use.selectedWallets();
  const showWalletsOnInit = !useBestRouteStore.use.walletsConfirmed();
```